api/src/main/java/org/openmrs/liquibase/ChangeLogDetective.java

```diff
@@ -157,13 +161,22 @@ boolean isVintageChangeSet(String filename, ChangeSet changeSet) {
 		}
 		return false;
 	}
-	
-	private void logUnrunChangeSetDetails(String filename, List<ChangeSet> filteredUnrunChangeSets) {
-		if (filteredUnrunChangeSets.size() < MAX_NUMBER_OF_CHANGE_SETS_TO_LOG) {
-			for (ChangeSet changeSet : filteredUnrunChangeSets) {
-				log.debug("file '{}' contains un-run change set with id '{}' by author '{}'", filename, changeSet.getId(),
+
+	/**
+	 * Logs un-run change sets no more than a given number and only for the 1.9.x Liquibase snapshots.
+	 * 
+	 * @return a boolean value indicating whether the change sets were logged. The value is used for testing.
+	 */
+	boolean logUnRunChangeSetDetails( String filename, List<ChangeSet> ChangeSets) {
+		if (ChangeSets.size() < MAX_NUMBER_OF_CHANGE_SETS_TO_LOG
```

Comment:
I guess these are usually not many, from the performance perspective. Correct?

Reply:
Yes, I remembered our discussion about having extra logs at all and took care to keep the number small.

In the case of the reference application there were two un-run change sets only and the max number to be logged is 9. If, for some reason, all of 1.9.x went missing, we still get the error and a single log messages stating the number of un-run change sets.

It took a couple of hours to recreate the bug and confirm what went wrong with the reference application yesterday, this patch is buying as and the community time where we can respond faster to any more findings.

Reply:
Fair enough. 👍